pkg/arvo/sys/vane/ames.hoon

```diff
@@ -691,35 +691,65 @@
   --
 ::  +scry: dereference namespace
 ::
-::    The ones producing vases are expected to be used like this:
-::
-::    &tang [(sell .^(vase %a /=peer=/~zod)) ~]
-::
 ++  scry
   |=  [fur=(unit (set monk)) ren=@tas why=shop syd=desk lot=coin tyl=path]
   ^-  (unit (unit cage))
-  ?.  =(lot [%$ %da now])  ~
-  ?.  =(%$ ren)  [~ ~]
-  ?.  =([%& our] why)
-    [~ ~]
-  ?+    syd  ~
-      %peers
-    ?^  tyl  [~ ~]
+  ::  only respond for the local identity, %$ desk, current timestamp
+  ::
+  ?.  ?&  =(&+our why)
+          =([%$ %da now] lot)
+          =(%$ syd)
+      ==
+    ~
+  ::  /ax/protocol/version           @
+  ::  /ax/peers                      (map ship ?(%alien known))
+  ::  /ax/peers/[ship]               ship-state
+  ::  /ax/peers/[ship]/forward-lane  (unit lane)
+  ::  /ax/bones/[ship]               [snd=(set bone) rcv=(set bone)]
+  ::  /ax/snd-bones/[ship]/[bone]    vase
+  ::
+  ?.  ?=(%x ren)  ~
+  ?+    tyl  ~
+      [%protocol %version ~]
+    ``noun+!>(protocol-version)
+  ::
+      [%peers ~]
     :^  ~  ~  %noun
     !>  ^-  (map ship ?(%alien %known))
     (~(run by peers.ames-state) head)
   ::
-      %peer
-    ?.  ?=([@ ~] tyl)  [~ ~]
-    =/  who  (slaw %p i.tyl)
+      [%peers @ *]
+    =/  who  (slaw %p i.t.tyl)
     ?~  who  [~ ~]
     ?~  peer=(~(get by peers.ames-state) u.who)
       [~ ~]
-    ``noun+!>(u.peer)
+    ?+  t.t.tyl  [~ ~]
+      ~  ``noun+!>(u.peer)
+    ::
+        [%forward-lane ~]
+      ::TODO  (list lane) instead?
```

Comment:
I'm concerned that a list of lanes could be used for amplification attacks.  Relay attacks are bad enough, but at least R0 isn't above 1.

Reply:
IMO a list of lanes is definitely going to be desirable at some point, mainly so you can say "i'm reachable at these different locations", where those locations might be different ip/port (in case of load balancer) or protocol than the outgoing.  R0 is still <= 1 as long as those destinations aren't themselves forwarding